cli/pub_command.go

```diff
@@ -301,88 +315,137 @@ func (c *pubCmd) doJetstream(nc *nats.Conn, progress *progress.Tracker) error {
 	return nil
 }
 
+// readLine reads a full line from a bufio.Reader regardless of buffer size,
+// this is important for reading lines longer than the bufio.Reader's default
+// buffer size and nicer than wholesale enlarging the buffer.
+func readLine(reader *bufio.Reader) (string, error) {
+	var buf bytes.Buffer
+
+	for {
+		line, isPrefix, err := reader.ReadLine()
+		buf.Write(line)
+
+		if err != nil || !isPrefix {
+			return buf.String(), err
+		}
+	}
+}
+
 func (c *pubCmd) publish(_ *fisk.ParseContext) error {
+	ctx, cancel := signal.NotifyContext(ctx, syscall.SIGTERM, syscall.SIGINT)
+	defer cancel()
+
 	nc, err := newNatsConn("", natsOpts()...)
 	if err != nil {
 		return err
 	}
 	defer nc.Close()
-
-	if c.cnt < 1 {
-		c.cnt = math.MaxInt16
-	}
-
-	if !c.bodyIsSet && (terminal.IsTerminal(int(os.Stdout.Fd())) || c.forceStdin) {
+	if !c.quiet {
 		log.Println("Reading payload from STDIN")
-		body, err := io.ReadAll(os.Stdin)
-		if err != nil {
-			return err
-		}
-		c.body = string(body)
 	}
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("interrupted")
+		default:
+			{
```

Comment:
not sure adding a block here adds any value other than make the code more annoying